pygtrie.py

```diff
@@ -147,60 +221,100 @@ def copy(self, make_copy, queue):
         return cpy
 
 
-class _Children(dict):
+class _Children(Children[_VT]):
     """Children collection representing more than one child."""
 
-    __slots__ = ()
+    __slots__ = ("data",)
 
-    def __init__(self, *items):
-        super().__init__(items)
+    def __init__(self, *items: Tuple[str, "_Node[_VT]"]):
+        self.data = dict(items)
+
+    def __bool__(self) -> bool:
+        return bool(self.data)
+
+    def __len__(self) -> int:
+        return len(self.data)
+
+    def items(self):
+        return self.data.items()
 
     def sorted_items(self):
         return sorted(self.items())
 
     def pick(self):
         return next(iter(self.items()))
 
-    def add(self, _parent, step):
-        self[step] = node = _Node()
+    def get(self, step: str) -> "_Node[_VT] | None":
+        return self.data.get(step)
+
+    def add(self, _parent: "_Node[_VT]", step: str):
+        self.data[step] = node = _Node()
         return node
 
-    def require(self, _parent, step):
-        return self.setdefault(step, _Node())
+    def require(self, _parent: "_Node[_VT]", step: str):
+        return self.data.setdefault(step, _Node())
 
     def merge(self, other, queue):
         """Moves children from other into this object."""
         for step, other_node in other.items():
-            node = self.setdefault(step, other_node)
+            node = self.data.setdefault(step, other_node)
             if node is not other_node:
                 queue.append((node, other_node))
         return self
 
     def delete(self, parent, step):
-        del self[step]
+        del self.data[step]
         if len(self) == 1:
-            parent.children = _OneChild(*self.popitem())
+            parent.children = _OneChild(*self.data.popitem())
 
     def copy(self, make_copy, queue):
         cpy = _Children()
-        cpy.update((make_copy(step), node.shallow_copy(make_copy))
-                   for step, node in self.items())
-        queue.append(cpy.values())
+        cpy.data.update(
+            (make_copy(step), node.shallow_copy(make_copy))
+            for step, node in self.items()
+        )
+        queue.append(tuple(cpy.data.values()))
         return cpy
 
 
-class _Node:
+V = TypeVar("V")
+V1 = TypeVar("V1")
+
+
+class _SentinelClass(enum.Enum):
+    _Sentinel = object()
+
+
+_SENTINEL: Final = _SentinelClass._Sentinel
+
+
+T_Iteritems = Callable[[Children[T]], Iterable[Tuple[str, "_Node[T]"]]]
+
+PathConv = Callable[[Tuple[str, ...]], str]
+NodeFactory = Callable[
+    [
+        Callable[[Tuple[str, ...]], str],
+        Tuple[str, ...],
+        Generator["_Node[T]", None, None],
+        Optional[T],
```

Comment:
This is backwards incompatible change and it makes it impossible to store None values in the trie.  The purpose of the factory being variadic was to have a way to detect whether node has value or it doesn’t.